tests/components/flo/test_device.py

```diff
@@ -88,13 +86,61 @@ async def test_device(
 
     assert aioclient_mock.call_count == call_count + 6
 
-    # test error sending device ping
+
+async def test_device_failures(
+    hass: HomeAssistant,
+    config_entry,
+    aioclient_mock_fixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test Flo by Moen devices buffer API failures."""
+    config_entry.add_to_hass(hass)
+    assert await async_setup_component(
+        hass, FLO_DOMAIN, {CONF_USERNAME: TEST_USER_ID, CONF_PASSWORD: TEST_PASSWORD}
+    )
+    await hass.async_block_till_done()
+    assert len(hass.data[FLO_DOMAIN][config_entry.entry_id]["devices"]) == 2
+
+    assert (
+        hass.states.get("sensor.smart_water_shutoff_current_system_mode").state
+        == "home"
+    )
+
+    aioclient_mock.clear_requests()
     with patch(
-        "homeassistant.components.flo.device.FloDeviceDataUpdateCoordinator.send_presence_ping",
+        "aioflo.presence.Presence.ping",
```

Comment:
Can we patch this at the place where it's used?

Reply:
it's used internally in the update coordinator

Reply:
Yes but when patching that's okay

Reply:
in the line it replaced :)

Reply:
> Yes but when patching that's okay

I have been told by Martin to avoid patching HA parts to patch the libs instead...

Reply:
I actually asked him last week why we prefer patches in HA over patching the whole lib and he responded https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Reply:
🤷🏻‍♂️  I put it back...